src/async_fuse/memfs/cache/memory_cache.rs

```diff
@@ -238,18 +398,108 @@ where
     }
 
     async fn flush(&self, ino: INum) {
+        if !self.write_through {
+            {
+                let mut pending_write_back = self.pending_write_back.write().await;
```

Comment:
So, do we need to hold the write lock during the IO time flushed out in a certain file?

Reply:
No, we don't. The write lock here is to ensure that no more blocks can be add to the `pending_write_back` queue, before we send a `Flush` command to the task. As we have sent the command, a file-level queue will be removed from `pending_write_back`, then the write lock can be released.